File: lecture11/exercise11-A/template.py
#!/usr/bin/env python3

# Exercise 11-A: Lotto

import itertools
import sys

# Constants

LOTTO_NUMBERS = 6

# Functions

def display_combinations(s, d, k):
    # TODO: Recursively display combinations

    # Base: have complete subset
    pass
    
    # Recurse: with current
    pass

    # Recurse: skip current
    pass

# Main execution

def main():
    for line, numbers in enumerate(map(str.split, sys.stdin)):
        if len(numbers) <= 1:
            break
        
        if line:
            print('')

        display_combinations([], list(map(int, numbers[1:])), 0)

if __name__ == '__main__':
    main()

File: lecture11/exercise11-A/template.cpp
// Exercise 11-A: Lotto

#include <algorithm>
#include <iostream>
#include <sstream>
#include <string>
#include <vector>

using namespace std;

// Constants

const int LOTTO_NUMBERS = 6;

// Functions

void display_combinations(vector<int> &s, vector<int> &d, size_t k) {
    // TODO: Recursively display combinations

    // Base: have complete subset
    
    // Recurse: with current

    // Recurse: skip current
}

// Main execution

int main(int argc, char *argv[]) {
    string line;
    for (size_t line_count = 0; getline(cin, line); line_count++) {
    	stringstream ss(line);
    	int k;
    	ss >> k;
    	if (!k) {
    	    break;
	}

	int n;
	vector<int> numbers;
	vector<int> combination;
	while (ss >> n) {
	    numbers.push_back(n);
	}
	
	sort(numbers.begin(), numbers.end());

	if (line_count) {
	    cout << endl;
	}

	display_combinations(combination, numbers, 0);
    }

    return 0;
}
